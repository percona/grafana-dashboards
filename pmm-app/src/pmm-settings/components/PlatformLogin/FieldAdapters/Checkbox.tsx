import React, { HTMLProps, useCallback } from 'react';
import { GrafanaTheme } from '@grafana/data';
import { stylesFactory, useTheme } from '@grafana/ui';
import { css, cx } from 'emotion';

export interface CheckboxProps extends Omit<HTMLProps<HTMLInputElement>, 'value'> {
  label?: string;
  value?: boolean;
}

export const getFocusCss = (theme: GrafanaTheme) => css`
  outline: 2px dotted transparent;
  outline-offset: 2px;
  box-shadow: 0 0 0 2px ${(theme.colors as any).pageBg}, 0 0 0px 4px ${theme.colors.formFocusOutline};
  transition: all 0.2s cubic-bezier(0.19, 1, 0.22, 1);
`;

export const getLabelStyles = stylesFactory((theme: GrafanaTheme) => ({
  label: css`
      font-size: ${theme.typography.size.sm};
      font-weight: ${theme.typography.weight.semibold};
      line-height: 1.25;
      margin: ${theme.spacing.formLabelMargin};
      padding: ${theme.spacing.formLabelPadding};
      color: ${theme.colors.formLabel};
      max-width: 480px;
    `,
}));

export const getCheckboxStyles = stylesFactory((theme: GrafanaTheme) => {
  const labelStyles = getLabelStyles(theme);
  const checkboxSize = '16px';

  return {
    label: cx(
      labelStyles.label,
      css`
        padding-left: ${theme.spacing.formSpacingBase}px;
      `
    ),
    wrapper: css`
      position: relative;
      padding-left: ${checkboxSize};
      vertical-align: middle;
    `,
    input: css`
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      opacity: 0;
      &:focus + span {
        ${getFocusCss(theme)}
      }
      &:checked + span {
        background: blue;
        background: ${theme.colors.formCheckboxBgChecked};
        border: none;
        &:hover {
          background: ${theme.colors.formCheckboxBgCheckedHover};
        }
        &:after {
          content: '';
          position: absolute;
          left: 5px;
          top: 1px;
          width: 6px;
          height: 12px;
          border: solid ${theme.colors.formCheckboxCheckmark};
          border-width: 0 3px 3px 0;
          transform: rotate(45deg);
        }
      }
    `,
    checkmark: css`
      display: inline-block;
      width: ${checkboxSize};
      height: ${checkboxSize};
      border-radius: ${theme.border.radius.sm};
      margin-right: ${theme.spacing.formSpacingBase}px;
      background: ${(theme.colors as any).formCheckboxBg};
      border: 1px solid ${theme.colors.formInputBorder};
      position: absolute;
      top: 1px;
      left: 0;
      &:hover {
        cursor: pointer;
        border-color: ${theme.colors.formInputBorderHover};
      }
    `,
  };
});

export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
  ({
    label, value, onChange, disabled, ...inputProps
  }, ref) => {
    const theme = useTheme();
    const handleOnChange = useCallback(
      (e: React.ChangeEvent<HTMLInputElement>) => {
        if (onChange) {
          onChange(e);
        }
      },
      [onChange]
    );
    const styles = getCheckboxStyles(theme);

    return (
      <label className={styles.wrapper}>
        <input
          type="checkbox"
          className={styles.input}
          checked={value}
          disabled={disabled}
          onChange={handleOnChange}
          {...inputProps}
          ref={ref}
        />
        <span className={styles.checkmark} />
        {label && <span className={styles.label}>{label}</span>}
      </label>
    );
  }
);

Checkbox.displayName = 'Checkbox';
