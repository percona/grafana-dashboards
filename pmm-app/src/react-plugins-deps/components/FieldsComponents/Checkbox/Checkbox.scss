#antd {
  .checkbox-container {
    --checkbox-size: 14px;
    --checkbox-border: rgb(123, 123, 123);
    --checkbox-color: rgb(211, 211, 211);

    position: relative;
    cursor: pointer;

    display: grid;
    grid-template-columns: auto 1fr;
    grid-column-gap: 10px;

    text-align: left;

    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
  }

  /* Styles for main checkbox*/
  .checkbox-container--main {
    font-weight: 700;
  }

  /* Styles for main (first) checkbox when it is unchecked*/
  .checkbox-container__main-input ~ span {
    opacity: 0.6;
  }

  /* Styles for main (first) checkbox when it is checked*/
  .checkbox-container__main-input:checked ~ span {
    opacity: 1;
  }

  /* Hide the browser's default checkbox */
  .checkbox-container input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
  }

  /* Create a custom checkbox */
  .checkbox-container__checkmark {
    width: 20px;
    height: 20px;

    color: var(--checkbox-color);
    border: 1px solid var(--checkbox-border);
    background: var(--page-background);
  }

  /* Create the checkmark/indicator (hidden when not checked) */
  .checkbox-container__checkmark:after {
    content: '';

    display: none;
  }

  /* Show the checkmark when checked */
  input:checked ~ .checkbox-container__checkmark:after {
    display: block;
  }

  /* Style the checkmark/indicator */
  .checkbox-container__checkmark:after {
    margin: 0 auto;
    width: calc(var(--checkbox-size) / 2);
    height: var(--checkbox-size);

    transform: rotate(45deg);
    border: solid var(--main-text-color);
    border-width: 0 2px 2px 0;
  }

  /* Styles for main (first) checkbox when it is unchecked*/
  .checkbox-container__main-input ~ span {
    opacity: 0.6;
  }

  /* Styles for main (first) checkbox when it is checked*/
  .checkbox-container__main-input:checked ~ span {
    opacity: 1;
  }

  /* Disabled styles */
  input:disabled ~ .checkbox-container__checkmark,
  input:disabled ~ .checkbox-container__label-text {
    opacity: 0.6;
    color: white;
  }

  .checkbox-container__label-text {
    word-break: break-all;
  }
}
