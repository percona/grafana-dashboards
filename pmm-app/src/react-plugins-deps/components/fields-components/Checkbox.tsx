import React, {FunctionComponent} from "react";
import { Field } from "react-final-form";

const CheckboxComponent = ({
  input,
  meta,
  required,
  labelCol,
  wrapperCol,
  ...rest
}) => {
  return (
    <input
      onChange={input.onChange}
      type="checkbox"
      className="input-field--dark"
    />
  );
};

interface CheckboxFieldInterface {
  required?: boolean;
  name: string;
  "data-cy": string;
}
export const CheckboxField = ({
  name,
  required,
  ...rest
}: CheckboxFieldInterface) => {
  return (
    <Field
      {...rest}
      name={name}
      component={CheckboxComponent as FunctionComponent}
      required={required}
    />
  );
};
