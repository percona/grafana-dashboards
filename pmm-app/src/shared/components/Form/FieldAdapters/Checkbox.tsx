import React, { HTMLProps, useCallback } from 'react';
import { GrafanaTheme2 } from '@grafana/data';
import { useStyles2 } from '@grafana/ui';
import { css, cx } from '@emotion/css';

export interface CheckboxProps extends Omit<HTMLProps<HTMLInputElement>, 'value'> {
  label?: string;
  value?: boolean;
}

export const getFocusCss = ({ v1: { colors } }: GrafanaTheme2) => css`
  outline: 2px dotted transparent;
  outline-offset: 2px;
  box-shadow: 0 0 0 2px ${(colors as any).pageBg}, 0 0 0px 4px ${colors.formFocusOutline};
  transition: all 0.2s cubic-bezier(0.19, 1, 0.22, 1);
`;

export const getLabelStyles = ({ v1: { typography, spacing, colors } }: GrafanaTheme2) => ({
  label: css`
      font-size: ${typography.size.sm};
      font-weight: ${typography.weight.semibold};
      line-height: 1.25;
      margin: ${spacing.formLabelMargin};
      padding: ${spacing.formLabelPadding};
      color: ${colors.formLabel};
      max-width: 480px;
    `,
});

export const getCheckboxStyles = (theme: GrafanaTheme2) => {
  const labelStyles = getLabelStyles(theme);
  const checkboxSize = '16px';

  return {
    label: cx(
      labelStyles.label,
      css`
        padding-left: ${theme.v1.spacing.formSpacingBase}px;
      `,
    ),
    wrapper: css`
      position: relative;
      padding-left: ${checkboxSize};
      vertical-align: middle;
    `,
    input: css`
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      opacity: 0;
      &:focus + span {
        ${getFocusCss(theme)}
      }
      &:checked + span {
        background: blue;
        background: ${theme.colors.primary.main};
        border: none;
        &:hover {
          background: ${theme.colors.primary.shade};
        }
        &:after {
          content: '';
          position: absolute;
          left: 5px;
          top: 1px;
          width: 6px;
          height: 12px;
          border: solid ${theme.colors.primary.contrastText};
          border-width: 0 3px 3px 0;
          transform: rotate(45deg);
        }
      }
    `,
    checkmark: css`
      display: inline-block;
      width: ${checkboxSize};
      height: ${checkboxSize};
      border-radius: ${theme.v1.border.radius.sm};
      margin-right: ${theme.v1.spacing.formSpacingBase}px;
      background: ${(theme.v1.colors as any).formCheckboxBg};
      border: 1px solid ${theme.v1.colors.formInputBorder};
      position: absolute;
      top: 1px;
      left: 0;
      &:hover {
        cursor: pointer;
        border-color: ${theme.v1.colors.formInputBorderHover};
      }
    `,
  };
};

export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
  ({
    label, value, onChange, disabled, ...inputProps
  }, ref) => {
    const handleOnChange = useCallback(
      (e: React.ChangeEvent<HTMLInputElement>) => {
        if (onChange) {
          onChange(e);
        }
      },
      [onChange],
    );
    const styles = useStyles2(getCheckboxStyles);

    return (
      <label className={styles.wrapper}>
        <input
          type="checkbox"
          className={styles.input}
          checked={value}
          disabled={disabled}
          onChange={handleOnChange}
          {...inputProps}
          ref={ref}
        />
        <span className={styles.checkmark} />
        {label && <span className={styles.label}>{label}</span>}
      </label>
    );
  },
);

Checkbox.displayName = 'Checkbox';
