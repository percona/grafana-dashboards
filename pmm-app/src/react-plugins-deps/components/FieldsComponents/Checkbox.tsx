import React from 'react';
// import { useField } from 'react-final-form-hooks';
import './Checkbox.scss';
interface CheckboxFieldInterface {
  required?: boolean;
  name?: string;
  'data-cy'?: string;
  form?: any;
  label?: string;
}
export const CheckboxField = ({ name, label, required, form, ...rest }: CheckboxFieldInterface) => {
  // const field = useField(name, form);

  return (
    <label className="checkbox-container checkbox-container--main">
      <input type="checkbox" />
      <span className="checkbox-container__checkmark"></span>
      <span className="checkbox-container__label-text">{label}</span>
    </label>
  );
};
