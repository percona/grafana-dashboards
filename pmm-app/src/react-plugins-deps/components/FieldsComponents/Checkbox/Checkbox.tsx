import React from 'react';
import { useField } from 'react-final-form-hooks';
import './Checkbox.scss';
// import { useField } from 'react-final-form';
interface CheckboxFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  form?: any;
  label?: string;
  checked?: boolean;
}
export const CheckboxField = ({ name, label, required, form, ...rest }: CheckboxFieldInterface) => {
  const field = useField(name, form);

  return (
    <label className="checkbox-container checkbox-container--main">
      <input {...field.input} type="checkbox" checked={field.input.value} />
      <span className="checkbox-container__checkmark"></span>
      <span className="checkbox-container__label-text">{label}</span>
    </label>
  );
};
