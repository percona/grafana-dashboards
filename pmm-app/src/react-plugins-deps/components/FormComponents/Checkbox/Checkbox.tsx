import React from 'react';
import { useField } from 'react-final-form';
import './Checkbox.scss';

interface CheckboxFieldInterface {
  required?: boolean;
  name: string;
  'data-cy'?: string;
  label?: string;
  checked?: boolean;
}

export const CheckboxField = ({ name, label, required }: CheckboxFieldInterface) => {
  const field = useField(name);

  return (
    <label className="checkbox-container checkbox-container--main">
      <input {...field.input} type="checkbox" checked={field.input.value} required={required} />
      <span className="checkbox-container__checkmark"></span>
      <span className="checkbox-container__label-text">{label}</span>
    </label>
  );
};
