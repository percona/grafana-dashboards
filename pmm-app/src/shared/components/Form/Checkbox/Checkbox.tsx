import React from 'react';
import { useField } from 'react-final-form';
import './Checkbox.scss';

interface CheckboxFieldInterface {
  required?: boolean;
  name: string;
  dataQa?: string;
  label?: string;
  checked?: boolean;
  disabled?: boolean;
}

export const CheckboxField = ({
  name, label, required, disabled, dataQa
}: CheckboxFieldInterface) => {
  const field = useField(name, { type: 'checkbox' });

  return (
    <label className="checkbox-container checkbox-container--main">
      <input
        {...field.input}
        type="checkbox"
        required={required}
        disabled={disabled}
        data-qa={dataQa}
      />
      <span className="checkbox-container__checkmark" />
      <span className="checkbox-container__label-text">{label}</span>
    </label>
  );
};
