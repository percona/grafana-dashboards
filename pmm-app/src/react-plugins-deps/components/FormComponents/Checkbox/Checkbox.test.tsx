// Just a stub test
import { CheckboxField } from './Checkbox';
import React from 'react';
import renderer from 'react-test-renderer';
import { renderHook } from '@testing-library/react-hooks';
import { Form } from 'react-final-form';

describe('Checkbox field test', () => {
  it('Checkbox renders correct without props', () => {
    renderHook(() => {
      const component = renderer.create(
        <Form onSubmit={() => null} render={() => <CheckboxField name="test_field" />} />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
