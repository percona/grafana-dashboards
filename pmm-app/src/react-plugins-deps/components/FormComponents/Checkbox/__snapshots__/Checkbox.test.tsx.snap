// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox field test Checkbox renders correct without props 1`] = `null`;
